Add ability to remove items from cart

scottfirestone opened the pull request:
-Added destroy route and controller action for removing a cart item.
-Added remove button to carts view, uses new cart method to remove item.
-Refactored cart index to use newly created model method that returns array of album objects.

scottfirestone committed:
Add ability to remove items from cart

-Added destroy route and controller action for removing a cart item.
-Added remove button to carts view, uses new cart method to remove item.
-Refactored cart index to use newly created model method that returns array of album objects.

scottfirestone commented:
@julsfelic @Salvi6God @Draganovic 
We did some refactoring for the carts index and have remove item functionality tested, working. Take a look!

ghost commented:
I've noticed some hound errors but other than that it should be good to merge! 

What do you think @julsfelic?

julsfelic commented on app/views/carts/index.html.erb:
```diff
@@ -1,12 +1,12 @@
+  <%= album.description %>
+  <%= album.formatted_price %>
+  <%= button_to "Remove", {:controller => :carts,
+      :action => 'destroy', :id => album.id }, :method => :delete %>
```
Maybe instead of having this logic in the view we could create a custom route that handles this for us?

``` ruby
  <%= button_to "Remove", carts_path(album.id), method: :delete %>
```

julsfelic commented on config/routes.rb:
```diff
@@ -4,7 +4,9 @@
-  resources :carts, only: [:index, :create]
+  resources :carts, only: [:index, :create, :destroy]
+
+  post "/carts", to: "carts#delete_item"
```
Check if left behind

julsfelic commented on app/controllers/cart_items_controller.rb:
```diff
@@ -0,0 +1,5 @@
+  def destroy
+    Cart.contents.delete(params[:id])
+  end
+end
```
Check to see if this is not needed

Draganovic committed:
Remove cart items controller

Cart items controller was not being used
Cart items resources were removed from the routes because they were not being used

julsfelic commented:
Good to go! I'll merge it in!
